src/evo/deterministicmns.h

```diff
@@ -193,27 +172,35 @@ class CDeterministicMNList
     void Serialize(Stream& s) const
     {
         const_cast<CDeterministicMNList*>(this)->SerializationOpBase(s, CSerActionSerialize());
+
         // Serialize the map as a vector
         WriteCompactSize(s, mnMap.size());
-        for (const auto& p : mnMap) {
-            s << *p.second;
+        for (const auto& [_, dmn] : mnMap) {
+            s << *dmn;
         }
     }
 
     template<typename Stream>
-    void Unserialize(Stream& s, const uint8_t format_version = CDeterministicMN::MN_CURRENT_FORMAT) {
-        mnMap = MnMap();
-        mnUniquePropertyMap = MnUniquePropertyMap();
-        mnInternalIdMap = MnInternalIdMap();
+    void Unserialize(Stream& s)
+    {
+        Clear();
 
         SerializationOpBase(s, CSerActionUnserialize());
 
-        size_t cnt = ReadCompactSize(s);
-        for (size_t i = 0; i < cnt; i++) {
-            AddMN(std::make_shared<CDeterministicMN>(deserialize, s, format_version), false);
+        size_t len{0};
+        len = ReadCompactSize(s);
+        for (size_t i = 0; i < len; i++) {
+            AddMN(std::make_shared<CDeterministicMN>(deserialize, s), /*fBumpTotalCount=*/false);
         }
     }
 
+    void Clear()
+    {
+        mnMap = MnMap();
+        mnUniquePropertyMap = MnUniquePropertyMap();
+        mnInternalIdMap = MnInternalIdMap();
```

Comment:
but not 
    uint256 blockHash;
    int nHeight{-1};
    uint32_t nTotalRegisteredCount{0};
?

Reply:
`Clear()` was extracted from deserialization routine (see below) for clarity but now clears all members. Resolved in latest push.

https://github.com/dashpay/dash/blob/e3b88ccefca66df55d57b66380a70bfb8e80cabe/src/evo/deterministicmns.h#L203-L208